### DISCUSSION TITLE
Very complicated and confusing...

### DISCUSSION THREAD

**Discussion by erikdemarco** [vikejs/vike#1850] [Discussion]
At first vike caught my attention with claiming to focuses on being an excellent frontend framework while not interfering with the rest of your stack and follows what we call a "do-one-thing-do-it-well architecture": So i click the link to their homepage.

Then read the quick start guide:

```
https://vike.dev/add
https://vike.dev/render-modes

Choose between SSR / pre-rendering (SSG) / SPA / HTML-only.
With Server Routing or Client Routing.
```

First they said they are frontend olny, now they want me to choose betwwen SPA, SSR, SSG, HTML-only. what is the difference between SSG and HTML-only? if there is SPA, why there is no MPA?

Then I also need to choose With Server Routing or Client Routing. What is that?? SSR is Server. SSG is Client. Why after i choose either SSR, or SSG, i need to choose again between client route or server route? But Ok lets continue.

My goal is simple. to do static site export. in nextjs its very simple just enable it from single config file. 
```
const nextConfig = {
  output: 'export',
}
```
 then it will export the static files, each routes gets its own directory.

Lets see how in vike, I choose SSG, so I follow the steps here:
```
1. deploy stackblitz using this template:
https://github.com/vikejs/vike/tree/main/examples/react-minimal

2. then follow this steps:
https://vike.dev/pre-rendering

3. do "npm run build"
```
And now in my dist directory, i have client&server folder. why? I just want SSG. How to use this splitted environment??

I tried all different configs combination:

```
// vite.config.js
import vike from 'vike/plugin'
export default {
  plugins: [
    vike({ prerender: true })
  ]
}

// /renderer/+config.js
// Environment: config
export default {
  meta: {
    Page: {
      env: { server: false, client: true } // SPA for all pages
    }
  }
}

// /pages/+config.js
export default {
  // Applies to all pages
  ssr: false
}
```

all of it failed, it still build both client and server files. and why vike has many different configs? so confusing and unorganized.

**Comment by brillout** [comment]
> i have client&server folder. why?

Indeed that's confusing. I just created https://github.com/vikejs/vike/issues/1851.

> all of it failed, it still build both client and server files. and why vike has many different configs? so confusing and unorganized.

You aren't supposed to use `meta` unless you know what you're doing. Same for manually implementing Render Modes. [I've updated the docs to reflect that](https://github.com/vikejs/vike/commit/b96cfc348270c8bd7065385c5d997d4e90218c32).

Let me know if you believe something else can be done to improve the docs's structure. Looking forward to more critical feedback.

As to what you're trying to achieve, why not simply using the `ssr` and `prerender` settings? These two settings cover what most users want to achieve in regards to render modes. Unless I'm missing something about what you're trying to achieve, it's a DX as trivial as Next.js's static export.
